### DISCUSSION TITLE
How can the final rotation of a sprite, with respect to the screen , be calculated?

### DISCUSSION THREAD

**Discussion by a-c-sreedhar-reddy** [pixijs/pixijs#7651] [Q&A]
In this case  a sprite is added to a container. The sprite is rotated by 15 degrees. and the container is rotated by 15 degrees. How can the global angle of the sprite be calculated (which would be 30) ?
```js
var app = new PIXI.Application({
    width: window.innerWidth,
    height: window.innerHeight,
    backgroundColor: 0x2c3e50
});
document.body.appendChild(app.view);

// Load the bunny texture
app.loader.add('bunny', 'https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg')
    .load(startup);

function startup()
{
    var bunny = new PIXI.Sprite(app.loader.resources.bunny.texture);
    bunny.angle = 15;

    const container = new PIXI.Container();
    container.angle = 15

    container.addChild(bunny)

    app.stage.addChild(container);

    //const bunnyAngle = bunny...; //30
    
}

```

**Comment by a-c-sreedhar-reddy** [comment]
I have tried the following function. But the sprite.worldTransform always return an identity matrix. [playground](https://www.pixiplayground.com/#/edit/KncJMklYROowViKmWVEou)
```js
function getRotation(sprite){
    
    console.log(sprite.worldTransform)
    console.log("a", sprite.worldTransform.a)

    const {a,b,c,d} = sprite.worldTransform;

    const skewX = -Math.atan2(-c, d);
    const skewY = Math.atan2(b, a);
    const delta = Math.abs(skewX + skewY);

    if (delta < 0.00001 || Math.abs(PI_2 - delta) < 0.00001)
    {
        return skewY
    }
    return 0;
}
```

**Reply by a-c-sreedhar-reddy** [reply]
Calling this function in the ticker gives the correct value.
https://www.pixiplayground.com/#/edit/EOt99ewLSs1MxfCYZNgCW

**Reply by ivanpopelyshev** [reply]
worldTransform is invalidated either in `updateTransform()` (whole subtree!) either in  `_recursivePostUpdateTransform()` (this element & parents)

Whenever you do worldTransform and you aren't sure whether it was calculated, you have to do that, like https://github.com/pixijs/pixijs/blob/c874bebfb7185efe44315ba3cbd50e36d13e94c2/packages/display/src/DisplayObject.ts#L675
